### ISSUE TITLE
image_index in a struct definition does not work properly on the HTML5 runtime

### ISSUE THREAD

**Issue by CST1229** [YoYoGames/GameMaker-Bugs#5968]
### Description
If you define an image_index variable on a struct *in its initial definition* (e.g `struct = {image_index: 123}`), on the HTML5 runtime/export/target/module trying to get it will not work and return undefined instead of the value you set. `image_index` is not a [struct forbidden variable](https://manual.gamemaker.io/monthly/en/GameMaker_Language/GML_Overview/Struct_Forbidden_Variables.htm) so it should work.
On other platforms (at least Windows, that's the one I tested) the bug does not occur, it will return the actual value as normal.
It also does not occur with some other built-in instance variables that are allowed in structs, like `sprite_index`. If you assign struct's `image_index` to some value afterwards, getting it will also now work on all platforms.

Small example script (also in the privately attached project):
```gml
var struct = {
	normal_var: 123,
	sprite_index: 456,
	image_index: 789,
};

// Both Windows and HTML5: prints 123
show_debug_message(struct.normal_var);
// Both Windows and HTML5: prints 456
show_debug_message(struct.sprite_index);
// Windows: prints 789
// HTML5: prints undefined
show_debug_message(struct.image_index);

struct.image_index = 101112;
// Both Windows and HTML5: prints 101112
show_debug_message(struct.image_index);
```

I assume this is a similar bug to https://github.com/YoYoGames/GameMaker-Bugs/issues/4874, but this does seem to be different (I can reproduce it on the 2024.4.1.202 runtime, which I'm pretty sure is after that bug was fixed).

### Steps To Reproduce
1. Create a struct with an image_index in its definition, like `struct = {image_index: 123}`
2. Try to get said image_index variable in any way, like `show_debug_message(struct.image_index)`
3. On the HTML5 runtime, it will return undefined, instead of the value you set
4. If you assign the struct's image_index afterwards, like `struct.image_index = 123`, or use a different variable name, getting will work

### Which version of GameMaker are you reporting this issue for?
IDE v2024.4.1.152 Runtime v2024.4.1.202

### Which operating system(s) are you seeing the problem on?
Windows 10.0.19045.0

### Which platform(s) are you seeing the problem on?
HTML5

[4350f9e4-5721-4c46-a2ad-5535ee05115b](https://api.gamemaker.io/api/github/downloads/4350f9e4-5721-4c46-a2ad-5535ee05115b)

**Comment by Grisgram** [comment]
i had the same problem when serializing/deserializing json for savegames.
in the end, I used `img_idx` when serializing/deserializing and mapped this name to the real image_index on the instance when it got restored.

Just posting this to backup your post - there _is_ something going on with `image_index` in html.

**Comment by rwkay** [comment]
Fixed in 2024.6 - obfuscator did not handle builtin instance variables well...

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.600.0.569  Runtime v2024.600.0.596
